src/PaymentGateways/Gateways/Stripe/CheckoutGateway.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe;
+
+use Give\Framework\PaymentGateways\Commands\GatewayCommand;
+use Give\Framework\PaymentGateways\Commands\PaymentProcessing;
+use Give\Framework\PaymentGateways\Commands\RedirectOffsite;
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\PaymentGateway;
+use Give\Helpers\Form\Utils as FormUtils;
+use Give\Helpers\Gateways\Stripe;
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give\PaymentGateways\Gateways\Stripe\Exceptions\CheckoutException;
+use Give\PaymentGateways\Gateways\Stripe\Helpers\CheckoutHelper;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\CheckoutSession;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentIntent;
+
+/**
+ * @unreleased
+ */
+class CheckoutGateway extends PaymentGateway
+{
+    use Traits\CheckoutInstructions;
+    use Traits\CheckoutModal;
+    use Traits\HandlePaymentIntentStatus;
+
+    /**
+     * @inheritDoc
+     * @unreleased
+     * @return GatewayCommand
+     * @throws PaymentGatewayException
+     */
+    public function createPayment( GatewayPaymentData $paymentData )
+    {
+        $workflow = new Workflow();
+        $workflow->bind( $paymentData );
+
+        $workflow->action( new Actions\SaveDonationSummary );
+        $workflow->action( new Actions\GetOrCreateStripeCustomer );
+
+        switch (give_stripe_get_checkout_type()) {
+            case 'modal':
+                return $this->createPaymentModal($workflow);
+            case 'redirect':
+                return $this->createPaymentRedirect($workflow);
+            default:
+                throw new CheckoutException('Invalid Checkout Error');
+        }
```

Comment:
@kjohnson just want to point out that actions should be as isolated as possible. I love your wizardry with the new workflow pattern, but it does kind of blurry the scope of each action because they all start to rely on the state of the container.  For instance, when I think of `GetOrCreateStripeCustomer` - all it should really be given is an email & potentially the donationId if you feel like updating the record in the same action - not necessarily the entire Gateway DTO.  

I know you know this, but for me when it comes to actions I always like to keep their scope as limited as possible so that I only have to give them a few simple params for testing.

Reply:
I've been considering revisiting the `Workflow`. The big benefit was to handle juggling passing dependencies around, but as I have refactored the code I've found that this may not be necessary any more. Most of the actions now only need `$paymentData`.

@jonwaldstein I'll refactor this when all the stripe gateways get merged, since refactoring out the `Workflow` class touches all of them.

Reply:
@kjohnson sounds good, that works for me - thanks!